homeassistant/components/recorder/history.py

```diff
@@ -124,10 +161,25 @@ def lambda_stmt_and_join_attributes(
     # without the attributes fields and do not join the
     # state_attributes table
     if no_attributes:
+        if schema_version >= 31:
```

Comment:
Since these all get compiled once we can't have any python inside the `lambda_stmt` which means we have to write them all out